creator/ingest_requests/models.py

```diff
@@ -1,16 +1,160 @@
+from django.contrib.auth import get_user_model
+from django.db.models.signals import pre_save
+from django.core.exceptions import ValidationError, FieldDoesNotExist
 from django.db import models
 
+from creator.files.models import Version
+from creator.jobs.models import Job
+
+DELIMITER = "-"
+NAME_PREFIX = "INGEST_REQUEST"
+
+User = get_user_model()
+
 
 class IngestRequest(models.Model):
     """
-    Plug the model in here
+    Request to ingest file(s) into a target data service
     """
 
     class Meta:
         permissions = [
             ("list_all_ingestrequest", "Show all ingest_requests"),
         ]
 
+    name = models.CharField(
+        max_length=500,
+        unique=True,
+        blank=True,
+        null=True,
+        help_text=(
+            "The name of the ingest request. An field that is auto populated "
+            "by concatenating the IDs of the IngestRequest's versions"
+        ),
+    )
+
+    versions = models.ManyToManyField(
+        Version,
+        related_name="ingest_requests",
+        help_text="List of files to ingest in the ingest request",
+    )
+
+    job = models.OneToOneField(
+        Job,
+        null=True,
+        blank=True,
+        related_name="ingest_request",
+        on_delete=models.CASCADE,
+        help_text="The associated ingest job for this ingest request",
+    )
+
     created_at = models.DateTimeField(
-        null=False, help_text="Time when the ingest_request was created"
+        auto_now_add=True,
+        null=True,
+        help_text="Time when the ingest request was created",
+    )
+
+    creator = models.ForeignKey(
+        User,
+        null=True,
+        blank=True,
+        related_name="ingest_requests",
+        on_delete=models.SET_NULL,
+        help_text="The user who submitted this ingest request",
     )
+
+    @staticmethod
+    def name_from_version_ids(version_ids):
+        """
+        Create IngestRequest.name by joining the version ids together
+        """
+        return DELIMITER.join([NAME_PREFIX] + version_ids)
```

Comment:
If you're trying to create a unique key for this ingest request that's reproducible, you probably want to sort your ids first.
I would suggest hashing the concatenated string instead, though, cause I assume there will be no limit on the number of versions in an ingest run.

Reply:
Yea I considered hashing it but I ultimately decided to keep it simple. I could consider that again though if we really want to keep name to a fixed length and store it more efficiently as a shorter string. 

True, there is no limit to the files that can be part of an ingest request (technically, we should be saying an ingest package) but realistically we will probably never ingest more than ~50 files in an ingest request which is why I just restricted name to a max length of 500 (11 char version id * ~50 files).